sonic-xcvrd/xcvrd/xcvrd_utilities/y_cable_helper.py

```diff
@@ -238,17 +238,16 @@ def y_cable_toggle_mux_torA(physical_port):
     port_instance = y_cable_port_instances.get(physical_port)
     if port_instance is None:
         helper_logger.log_error(
-            "Error: Could not get port instance for read side for  Y cable port {}".format(physical_port))
+            "Error: Could not get port instance for read side for  Y cable port {} {}".format(physical_port, threading.currentThread().getName()))
         return -1
 
-    with y_cable_port_locks[physical_port]:
-        try:
-            update_status = port_instance.toggle_mux_to_tor_a()
```

Comment:
toggling the mux is a write operation, would we still need the lock for this?

Reply:
This should be fine, as long as no firmware API's are getting called in sync we are good. 
as the read write eeprom also has a mutex lock in the platform api call, optoe implementation. 
https://github.com/Azure/sonic-linux-kernel/blob/889d76e36b3f012d3782a1c5e1587c32e4d1ed11/patch/driver-support-optoe.patch#L717

So theoretically we can get rid of all the locking. 
the only problem is firmware API's have a parallelism problem(in the firmware confirmed with vendor)
so that's why we have the locks still in place.

Reply:
Will this vary between different cable vendors?

Reply:
Both the vendors have the same ask currently.

Reply:
But would it be possible for a vendor to not include the mutex in their own implementation, which means we would need it here?

Reply:
mutex is provided by optoe layer, all vendors go through that read/write eeprom API